### PR TITLE
a.out helpers

### PR DESCRIPTION
**Pull request by brianyu28**
Addressing #58 and #60.

The name `a.out.py` ran into trouble when trying to import it into `application.py` because Python uses the `.` as a way to separate modules.

Tentatively renamed to `aout.py`, which I'm not convinced is the most elegant name for the file, thoughts on anything better?

### CODE REVIEW

File: helpers/aout.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        ]
+        return (1, response)
+    
+    # $ ./a.out
+    # Segmentation fault
+    matches = re.match("Segmentation fault", lines[0])
+    if matches:
+        response = [
+            "Looks like your program is trying to access areas of memory that it isn't supposed to access.",
+            "There are many causes for segmentation faults, but you may want to consider:",
```
Comment (dmalan) [inline helpers/aout.py]: Not sure we need this preamble, since many helpers already offer multiple suggestions?

File: helpers/aout.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            "Are you trying to access an element of an array beyond the size of the array?",
+            "Are you taking care to only dereference a pointer after you've initialized it (perhaps with `malloc`)?",
+            "Have you checked that any pointers you're dereferencing aren't NULL?",
```
Comment (dmalan) [inline helpers/aout.py]: \`NULL\`

File: helpers/aout.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    
+    # $ ./a.out
+    # Segmentation fault
+    matches = re.match("Segmentation fault", lines[0])
+    if matches:
+        response = [
+            "Looks like your program is trying to access areas of memory that it isn't supposed to access.",
+            "There are many causes for segmentation faults, but you may want to consider:",
+            "Are you trying to access an element of an array beyond the size of the array?",
+            "Are you taking care to only dereference a pointer after you've initialized it (perhaps with `malloc`)?",
```
Comment (dmalan) [inline helpers/aout.py]: This seems the opposite of what we want to say? "Are you dereferencing a pointer that you haven't initialized?"

File: helpers/aout.py (hunk 2 above)
Comment (dmalan) [inline helpers/aout.py]: I think we should focus on what the code is doing, not the student, so to speak. E.g., "Are you dereferencing a pointer whose value is \`NULL\`?"

File: helpers/aout.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+            "Have you checked that any pointers you're dereferencing aren't NULL?",
+            "Are you trying to dereference a pointer after you've freed it?"
```
Comment (dmalan) [inline helpers/aout.py]: s/trying to//

**Comment by dmalan** [comment]
Not opposed to just `a.py`!

**Comment by brianyu28** [comment]
Fixed.

**Comment by dmalan** [comment]
:shipit:

**Pull request merged**